server/src/main/java/io/crate/expression/scalar/HasSchemaPrivilegeFunction.java

```diff
@@ -270,20 +271,29 @@ private Collection<Privilege.Type> normalizePrivilegeIfLiteral(Symbol symbol) {
     public final Boolean evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] args) {
         Object userNameOrOid, schemaNameOrOid, privileges;
 
+        var sessionUser = USER_BY_NAME.apply(nodeCtx.userLookup(), txnCtx.sessionSettings().userName());
+        User user;
         if (args.length == 2) {
-            userNameOrOid = txnCtx.sessionSettings().userName();
             schemaNameOrOid = args[0].value();
             privileges = args[1].value();
+            user = sessionUser;
         } else {
             userNameOrOid = args[0].value();
+            if (userNameOrOid == null) {
+                return null;
+            }
+            user = getUser.apply(nodeCtx.userLookup(), userNameOrOid);
+            // Only superusers can call this function for other users
```

Comment:
I think in most cases compiled version will be used, so we need similar check in the end of the `compile` method.

Reply:
Thank you!